core/templates/dev/head/domain/topic/TopicRightsBackendApiService.js

```diff
@@ -0,0 +1,145 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service to change the rights of topic in the backend.
+ */
+
+oppia.factory('TopicRightsBackendApiService', [
+  '$http', '$log', '$q', 'UrlInterpolationService',
+  'TOPIC_MANAGER_RIGHTS_URL_TEMPLATE', 'TOPIC_RIGHTS_URL_TEMPLATE',
+  function($http, $log, $q, UrlInterpolationService,
+      TOPIC_MANAGER_RIGHTS_URL_TEMPLATE, TOPIC_RIGHTS_URL_TEMPLATE) {
+    // Maps previously loaded topic rights to their IDs.
+    var topicRightsCache = {};
+
+    var _fetchTopicRights = function(topicId, successCallback,
+        errorCallback) {
+      var topicRightsUrl = UrlInterpolationService.interpolateUrl(
+        TOPIC_RIGHTS_URL_TEMPLATE, {
+          topic_id: topicId
+        });
+
+      $http.get(topicRightsUrl).then(function(response) {
+        if (successCallback) {
+          successCallback(response.data.topic_rights);
+        }
+      }, function(errorResponse) {
+        if (errorCallback) {
+          errorCallback(errorResponse.data);
+        }
+      });
+    };
+
+    var _setTopicStatus = function(
+        topicId, publishStatus, successCallback, errorCallback) {
+      var changeTopicStatusUrl = UrlInterpolationService.interpolateUrl(
+        '/rightshandler/change_topic_status/<topic_id>', {
+          topic_id: topicId
+        });
+
+      var putParams = {
+        publish_status: publishStatus
+      };
+
+      $http.put(changeTopicStatusUrl, putParams).then(function(response) {
+        topicRightsCache[topicId] = response.data;
+        if (successCallback) {
+          successCallback(response.data);
+        }
+      }, function(errorResponse) {
+        if (errorCallback) {
+          errorCallback(errorResponse.data);
+        }
+      });
+    };
+
+    var _isCached = function(topicId) {
+      return topicId in topicRightsCache;
```

Comment:
`return topicRightsCache.hasOwnProperty(topicId);`

(Btw, for these comments in general, worth changing the source too if this is based on something else in the codebase. Better to make it all consistent.)

Reply:
Done.